File: api/models/openai.go
package models

const GPT_API_URL = "https://api.openai.com/v1/chat/completions"

type GptApiRequest struct {
	Model       string       `json:"model"`
	Messages    []GptMessage `json:"messages"`
	Temperature float64      `json:"temperature"`
}

type GptApiResponse struct {
	Choices []gptChoice `json:"choices"`
}

type GptMessage struct {
	Role    string `json:"role"`
	Content string `json:"content"`
}

type gptChoice struct {
	Message GptMessage `json:"message"`
}

type GptModel string

const (
	GPT_MODEL_3_5_TURBO       = "gpt-3.5-turbo"
	GPT_MODEL_4_TURBO_PREVIEW = "gpt-4-1106-preview"
)

File: src/models/openai.ts
export const GPT_API_URL = "https://api.openai.com/v1/chat/completions";

export interface GptApiRequest {
  model: string;
  messages: GptMessage[];
  temperature: number;
}

export interface GptMessage {
  role: string;
  content: string;
}

export interface GptApiResponse {
  choices: GptChoice[];
}

export interface GptChoice {
  message: GptMessage;
}

export enum GptModel {
  GPT_5_2 = "gpt-5.2",
  GPT_5_1 = "gpt-5.1",
  GPT_5 = "gpt-5",
  GPT_5_MINI = "gpt-5-mini",
  GPT_5_NANO = "gpt-5-nano"
}
